### PR TITLE
Create capability to create cookies that expire when browser is close…

### PR DESCRIPTION
**Pull request by jgrayauskie**
…d / session cookie

* Allow for expire to be null, refactor expiration check
* remove max age option if set to ‘session’
* keep backward compatibility
* update read me
* add test

### CODE REVIEW

**Comment by jgrayauskie** [comment]
#76

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "koa-session",
   "description": "Koa cookie session middleware",
   "repository": "koajs/session",
-  "version": "4.0.1",
+  "version": "4.0.2",
```
Comment (dead-horse) [inline package.json]: don't change version in PR

File: test/store.test.js [hunk 2]
```diff
@@ -356,6 +356,34 @@ describe('Koa Session External Store', () => {
...
+          this.body = this.session.message;
+        });
+        const server = app.listen();
+
+        request(server)
+        .post('/')
+        .expect('Set-Cookie', /koa:sess/)
+        .end((err, res) => {
+          if (err) return done(err);
+          const cookie = res.headers['set-cookie'].join(';');
```
Comment (dead-horse) [inline test/store.test.js]: need an assertion to check if maxAge set

File: Readme.md [hunk 3]
```diff
@@ -47,7 +47,10 @@ app.keys = ['some secret hurr'];
 
 var CONFIG = {
   key: 'koa:sess', /** (string) cookie key (default is koa:sess) */
-  maxAge: 86400000, /** (number) maxAge in ms (default is 1 days) */
+  /** (number) maxAge in ms (default is 1 days) */
+    maxAge: 86400000,
+  /** string: 'session' will result in a cookie that expires when session/browser is closed */
+    maxAge: 'session',
```
Comment (dead-horse) [inline Readme.md]: must add a warning message here: if set maxAge to `session`, once the cookie was stolen, the cookie will never expired.

File: package.json (hunk 1 above)
Comment (jgrayauskie) [inline package.json]: 👍

File: lib/context.js [hunk 4]
```diff
@@ -242,13 +242,18 @@ class ContextSession {
...
-    const maxAge = opts.maxAge || ONE_DAY;
-
     let json = this.session.toJSON();
     // set expire for check
-    json._expire = maxAge + Date.now();
-    json._maxAge = maxAge;
+    const maxAge = opts.maxAge ? opts.maxAge : ONE_DAY;
+    if (maxAge === 'session') {
+      // do not set _expire in json if maxAge is set to 'session'
+      // also delete maxAge from options
```
Comment (jgrayauskie) [inline lib/context.js]: 👍

File: test/store.test.js (hunk 2 above)
Comment (jgrayauskie) [inline test/store.test.js]: Good Call! just added one 👍

**Comment by kylehood** [comment]
Is there any other feedback pending on this PR?

**Comment by dead-horse** [comment]
LGTM

**Comment by dead-horse** [comment]
4.1.0 released, I'll cherry-pick this commit to master

**Pull request merged**